ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -559,11 +642,29 @@ class JITModBuilder {
           continue;
         }
 
-        if ((isa<BranchInst>(I)) || isa<SwitchInst>(I)) {
-          // FIXME Replace all potential CFG divergence with guards.
-          continue;
+        if (isa<BranchInst>(I)) {
+          // Should not be an unmappable hole in the trace after a branch.
```

Comment:
Do we know that `Idx + 1` is always within bounds?

Reply:
The bounds are checked with an assertion [here](https://github.com/vext01/yk_pv/blob/8b1144172b3123dff939624b1cbfa92700fe0b32/ykllvmwrap/src/jitmodbuilder.cc#L68).

Reply:
OK, so that `assert` rules out memory unsafety, but I think my question stands as an issue of correctness (as it would in e.g. Python): do we know for sure that there's always a `Idx + 1`?

Reply:
There must be another block in the trace following a branch, yes.

Reply:
OK.